### PR TITLE
CY-383: Add AD - Multiple Password Changes in Short Time Period alert

### PR DESCRIPTION
**Pull request by mahirchavda**
* Add AD - Multiple Password Changes in Short Time Period alert & filter macros
* Add important field in the AD - Password Change Outside Working Hour alert and panel

### CODE REVIEW

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 1]
```diff
@@ -934,14 +934,47 @@ search = `cs_wineventlog_security` EventCode IN (627, 628, 4723, 4724, 4794) \
...
+Data Collection - Splunk_TA_windows - WinEventLog://Security stanza
+dispatch.earliest_time = -62m@m
+dispatch.latest_time = -2m@m
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_wineventlog_security` EventCode IN (627, 628, 4723, 4724, 4794) \
+| table _time, ComputerName, Target_Account_Name, Target_Account_Domain, Subject_Account_Name Subject_Account_Domain Caller_Workstation, msad_action, action, EventCode \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:966]: Only one table (last one) command is required for data formatting.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 2]
```diff
@@ -934,14 +934,47 @@ search = `cs_wineventlog_security` EventCode IN (627, 628, 4723, 4724, 4794) \
...
+| stats count, latest(_time) as _time, values(*) as * by ComputerName, Target_Account_Name \
+| table _time, ComputerName, Target_Account_Name, count, Target_Account_Domain, Subject_Account_Name Subject_Account_Domain Caller_Workstation, msad_action, action, EventCode \
+| where count>=`cs_ad_multiple_password_changes_count_limit` \
+| sort - count \
+| `cs_ad_multiple_password_changes_filter`
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:971]: Use filter before sort command.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -934,14 +934,47 @@ search = `cs_wineventlog_security` EventCode IN (627, 628, 4723, 4724, 4794) \
...
+dispatch.latest_time = -2m@m
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_wineventlog_security` EventCode IN (627, 628, 4723, 4724, 4794) \
+| table _time, ComputerName, Target_Account_Name, Target_Account_Domain, Subject_Account_Name Subject_Account_Domain Caller_Workstation, msad_action, action, EventCode \
+| stats count, latest(_time) as _time, values(*) as * by ComputerName, Target_Account_Name \
+| table _time, ComputerName, Target_Account_Name, count, Target_Account_Domain, Subject_Account_Name Subject_Account_Domain Caller_Workstation, msad_action, action, EventCode \
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:968]: Use table command after where condition.

**Review by VatsalJagani — APPROVED** [review]
Just few query improvements.

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 1 above)
Comment (mahirchavda) [reply cyences_app_for_splunk/default/savedsearches.conf:966]: If we remove first table then stats will do values(*) for all the fields so to limit the fields first table the required fields. and second table is to show fields in the specific format.

Comment (VatsalJagani) [reply cyences_app_for_splunk/default/savedsearches.conf:966]: Use fields command.

**Review by VatsalJagani — APPROVED** [review]

**Pull request merged**